sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1839,6 +1839,16 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(false)
 
+  val ALLOW_MULTIPLE_JARS = conf("spark.rapids.sql.allowMultipleJars")
+    .internal()
+    .startupOnly()
+    .doc("Allow multiple rapids-4-spark, spark-rapids-jni, and cudf jars on the classpath. " +
+      "Spark will take the first one it finds, so the version may not be expected. Possisble " +
+      "values are ALWAYS: allow all jars, SAME_REVISION: only allow jars with the same " +
+      "revision, NEVER: do not allow multiple jars at all.")
+    .stringConf
+    .createWithDefault("SAME_REVISION")
```

Comment:
I still think NEVER should be the default, and SAME_REVISION should only be set in the tests where it is hard to fix right away with an issue filed to get it fixed later.

Reply:
Ok, will file a follow-on issue on it. 

What are the possible issues if there are multiple jars of the same revision on the classpath?

Reply:
Here are some problems going undetected:
- it does not detect when the dist jar is in multiple extraClassPath, $SPARK_HOME/jars, and on the user classpath via packlages and jars. We have seen bugs in such deployments #5796
- when you have [sql-plugin and aggregator on the classpath at the same time](https://github.com/NVIDIA/spark-rapids/pull/9654#issuecomment-1811693631) the user might be using an unintended flatbffers dependency, the 
- aggregator relocate flatbuffers to an internal package to make sure it uses just the one from the aggregator.

It is fine to have a separate issue to sort the right default value out.